### PR TITLE
{Network} `az network bastion`: Changes to accommodate MFA auth

### PR DESCRIPTION
**Pull request by aavalang**
**Related command**
az network bastion

**Description**
This code is now using a onetime token to establish weboscket connection to the target VM as opposed to a static token before

**Testing Guide**
tested with new bastions in prod environment

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Network

File: src/azure-cli/azure/cli/command_modules/network/_params.py [hunk 1]
```diff
@@ -2149,6 +2149,7 @@ def load_arguments(self, _):
         c.argument('ssh_key', help='SSH key file location for SSH connections.', options_list=['--ssh-key'])
     with self.argument_context('network bastion rdp') as c:
         c.argument('disable_gateway', arg_type=get_three_state_flag(), help='Flag to disable access through RD gateway.')
+        c.argument('enable_mfa', help='Enable RDS auth for MFA if supported by the target machine.')
```
Comment (necusjz) [inline src/azure-cli/azure/cli/command_modules/network/_params.py:2152]: It's better to leverage https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#registering-flags.

Comment (AndrewRobie) [inline src/azure-cli/azure/cli/command_modules/network/_params.py:2152]: Do you need a min api version for this?

File: src/azure-cli/azure/cli/command_modules/network/tunnel.py [hunk 2]
```diff
@@ -103,9 +104,10 @@ def _get_auth_token(self):
                 exp = CloudError(response)
             raise exp
 
+        self.ws_token = response_json["websocketToken"]
```
Comment (AndrewRobie) [inline src/azure-cli/azure/cli/command_modules/network/tunnel.py:107]: It looks like this is only used once and that's through the return. Given that maybe we don't need to save it as a class variable.

**Review by AndrewRobie — DISMISSED** [review]

File: src/azure-cli/azure/cli/command_modules/network/tunnel.py (hunk 2 above)
Comment (aavalang) [reply src/azure-cli/azure/cli/command_modules/network/tunnel.py]: makes sense.

File: src/azure-cli/azure/cli/command_modules/network/_params.py (hunk 1 above)
Comment (aavalang) [reply src/azure-cli/azure/cli/command_modules/network/_params.py]: done

**Review by necusjz — APPROVED** [review]

**Pull request merged**